addons/auth_password_policy/static/src/password_policy.js

```diff
@@ -0,0 +1,110 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { sprintf } from "@web/core/utils/strings";
+
+export class Policy {
+    /**
+     * @param {String} password
+     * @returns {number}
+     */
+    score(password) {}
+}
+
+export class ConcretePolicy extends Policy {
+    /**
+     * @param {Object} info
+     * @param {Number} [info.minlength=0]
+     * @param {Number} [info.minwords=0]
+     * @param {Number} [info.minclasses=0]
+     */
+    constructor({minlength, minwords, minclasses}) {
+        super();
+        this.minlength = minlength || 0;
+        this.minwords = minwords || 0;
+        this.minclasses = minclasses || 0;
```

Comment:
detail: variables and properties are usually camel case in javascript, so `minLength`, `minWords`, etc.